search/result_processor.py

```diff
@@ -0,0 +1,147 @@
+        """ decorate the matches within the excerpt """
+        matches = re.finditer(match_word, match_in, re.IGNORECASE)
+        for matched_string in set([match.group() for match in matches]):
+            match_in = match_in.replace(
+                matched_string,
+                getattr(settings, "SEARCH_MATCH_DECORATION", u"<b>{}</b>").format(matched_string)
+            )
+        return match_in
+
+    # disabling pylint violations because overriders will want to use these
+    def should_remove(self, user):  # pylint: disable=unused-argument, no-self-use
+        """
+        Override this in a class in order to add in last-chance access checks to the search process
+        Your application will want to make this decision
+        """
+        return False
+
+    def add_properties(self):
+        """
+        Called during post processing of result
+        Any properties defined in your subclass will get exposed as members of the result json from the search
+        """
+        for property_name in [p[0] for p in inspect.getmembers(self.__class__) if isinstance(p[1], property)]:
+            self._results_fields[property_name] = getattr(self, property_name, None)
+
+    @classmethod
+    def process_result(cls, dictionary, match_phrase, user):
+        """
+        Called from within search handler. Finds desired subclass and decides if the
+        result should be removed and adds properties derived from the result information
+        """
+        result_processor = _load_class(getattr(settings, "SEARCH_RESULT_PROCESSOR", None), cls)
+        srp = result_processor(dictionary, match_phrase)
+        if srp.should_remove(user):
+            return None
+        srp.add_properties()
+        return dictionary
+
+    @property
+    def excerpt(self):
+        """
+        Property to display a useful excerpt representing the matches within the results
+        """
+        if "content" not in self._results_fields:
+            return None
+
+        match_words = [self._match_phrase]
+        separate_words = self._match_phrase.split(' ')
+        if len(separate_words) > 1:
+            match_words.extend(self._match_phrase.split(' '))
+
+        matches = SearchResultProcessor.find_matches(
+            SearchResultProcessor.strings_in_dictionary(self._results_fields["content"]),
+            match_words,
+            DESIRED_EXCERPT_LENGTH
+        )
+        excerpt_text = ELLIPSIS.join(matches)
+
+        for match_word in match_words:
+            excerpt_text = SearchResultProcessor.decorate_matches(excerpt_text, match_word)
+
+        return excerpt_text
+
+    @property
+    def url(self):
+        """
+        Property to display the url for the given location, useful for allowing navigation
+        """
+        if "course" not in self._results_fields or "id" not in self._results_fields:
+            return None
+
+        return u"/courses/{course_id}/jump_to/{location}".format(
```

Comment:
Yes, I completely agree; the LMS overrides this class for use from the search url handler, and this property should be defined there and not here. I shall move it.